Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/PlayerInput.cs

```diff
@@ -1930,6 +1927,31 @@ private void OnDeviceChange(InputDevice device, InputDeviceChange change)
             }
         }
 
+        private bool IsCurrentControlScheme(ref InputControlScheme scheme, InputDevice[] devices)
+        {
+            var isSameScheme = user.controlScheme.HasValue && user.controlScheme.Value.Equals(scheme);
+            return (isSameScheme &&
+                ArrayHelpers.EqualSets(user.pairedDevices.ToArray(), devices) &&
```

Comment:
Would `devices.All(user.pairedDevices.Contains)` do the same thing? It would avoid the new array allocation and all the extra code in EqualSets.

Reply:
Almost but creates some issues due to IReadOnlyArray and value types. Replaced by other comparison, see latest commit.

Reply:
I don't see it. Where is the value type in this? Seems like reference comparisons between devices will work well here, especially since the sample code in the documentation for SwitchCurrentControlScheme uses Keyboard.current and Mouse.current as the devices.

Reply:
Hopefully this makes more sense to you now @andrew-oc , thanks for the discussion. Also realized that pairing and controller scheme switch should be uncorrelated actions. Unfortunately not possible to write a test for the case of doing unnecessary work on pairing since pairing/unpairing doesn't seem to be observable, but verified manually with user-submitted test project in additional to the automated tests.